src/dataMigration.js

```diff
@@ -188,6 +188,46 @@ const dataMigrations = [
       });
     },
   },
+
+  //  2.3.6 Introduced a patch for ItemBatch outgoing sync supplier. Previously,
+  // all outgoing ItemBatches had their supplier field set to the default
+  // supplying store record for its store. 2.3.6 ensured that the supplier
+  // of a ItemBatch was correctly synced out, maintaining it's supplier field.
+  // This migration code
+  {
+    version: '3.0.0',
+    migrate: database => {
+      const supplierInvoiceType = 'supplier_invoice';
+      const finalisedStatus = 'finalised';
+      const itemBatches = database.objects('ItemBatch');
+      database.write(() => {
+        // For each item batch, find the supplier invoice transaction which
+        // introduced it and set the otherParty of the related transaction
+        // to the supplier of the related batch.
+        itemBatches.forEach(batch => {
+          const { id, transactionBatches } = batch;
+          // Ensure there are transaction batches before trying to find the transaction.
+          if (!(transactionBatches && transactionBatches.length !== 0)) return;
+          // There should only be one supplier invoice related to the item batch,
+          // in case somethings wrong and there are more, use the first one.
+          // Also ensure it is finalised. If it isn't, the supplier will be set when confirmed.
+          const supplierInvoiceTransactionBatches = transactionBatches.filtered(
+            'transaction.type = $0 && transaction.status = $1',
+            supplierInvoiceType,
+            finalisedStatus
+          );
+          // If there aren't any transaction batches, exit.
+          if (!(supplierInvoiceTransactionBatches.length > 0)) return;
+          const { transaction } = supplierInvoiceTransactionBatches[0];
+          // If somehow the transaction batch doesn't have a transaction, return
+          if (!transaction) return;
+          const { otherParty: supplier } = transaction;
+          if (!supplier) return;
+          database.update('ItemBatch', { id, supplier });
```

Comment:
can you do this only if otherParty has changed ?

Reply:
515fdd07f890f9b71d59cf6c5a3f7faf4df96984